cumulus/client/relay-chain-minimal-node/src/network.rs

```diff
@@ -114,6 +119,18 @@ pub(crate) fn build_collator_network(
 	Ok((network_service, network_starter, Box::new(SyncOracle {})))
 }
 
+fn adjust_network_config_light_in_peers(config: &mut NetworkConfiguration) {
+	let light_client_in_peers = (config.default_peers_set.in_peers +
+		config.default_peers_set.out_peers)
+		.saturating_sub(config.default_peers_set_num_full);
```

Comment:
Even though it looks ugly, it seems there is no other way to estimate `in_peers_light` using current implementation of `NetworkConfiguration`...

Reply:
Yes, this should probably become a proper field in the config some day.

Reply:
There is something I don't understand, maybe due to some limited domain knowledge in the networking subsystem. How reducing the `in_peers` number just trims the light clients connections?
I mean, aren't the full clients connections affected as well?

Reply:
This is a bit tricky, when users specify `--in-peers-light` we actually add them to the normal `in_peers` in config construction: https://github.com/paritytech/polkadot-sdk/blob/b44de38dde1a6d76131ba2c4123a500aa9e0b70e/substrate/client/cli/src/params/network_params.rs#L235-L245.

And here is the logic that checks for the light client slots actually occupied:
https://github.com/paritytech/polkadot-sdk/blob/b44de38dde1a6d76131ba2c4123a500aa9e0b70e/substrate/client/network/sync/src/engine.rs#L1035-L1041

The whole concept of light client slots is kind of implicit currently. But in the end you can still calculate the light client slots with  by calculating the difference as in the code.

Reply:
I was going to comment, but @skunert beat me. Could just add that it's the limitation of light peer slots bookkeeping, we calculate the slot count in `SyncingEngine` from `in_peers`, `out_peers`, `full_peers`:
https://github.com/paritytech/polkadot-sdk/blob/4caa3d8d8e6f90b2689cd2666c94044394712abf/substrate/client/network/sync/src/engine.rs#L422-L426
So it's enough to update `config.default_peers_set.in_peers` for light peer slots to be correctly calculated later.

Reply:
clear Ty